src/FillArrays.jl

```diff
@@ -73,6 +73,22 @@ convert(::Type{AbstractFill}, F::AbstractFill) = F
 convert(::Type{AbstractFill{T}}, F::AbstractFill) where T = convert(AbstractArray{T}, F)
 convert(::Type{AbstractFill{T,N}}, F::AbstractFill) where {T,N} = convert(AbstractArray{T,N}, F)
 
+""" Throws an error if `arr` does not contain one and only one unique value. """
+function unique_value(arr::AbstractArray)
+    if isempty(arr) error("Cannot convert empty array to Fill") end
+    val = first(arr)
+    for x in arr
+        if x !== val
+            error("Input array contains both $x and $val. Cannot convert to Fill")
+        end
+    end
+    return val
+end
+convert(::Type{Fill}, arr::AbstractArray{T}) where T = Fill{T}(unique_value(arr), axes(arr))
+convert(::Type{Fill{T}}, arr::AbstractArray) where T = Fill{T}(unique_value(arr), axes(arr))
+convert(::Type{T}, F::T) where T<:Fill = F   # ambiguity fix
```

Comment:
Add the following:
```julia
convert(::Type{Fill{T}}, F::AbstractFill) = Fill{T}(getindex_value(F), axes(F))
convert(::Type{Fill}, F::AbstractFill{T}) where T = Fill{T}(getindex_value(F), axes(F))
```

Reply:
Or just `unique_value(f::AbstractFill) = getindex_value(f)`?

Reply:
Done.